### PR TITLE
WIP: Added Jenkinlogger

### PR DESCRIPTION
**Pull request by codeasashu**
# Description

See [JENKINS-59766](https://issues.jenkins-ci.org/browse/JENKINS-59766).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description.
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests.
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description.
- [ ] Reviewed the code.
- [ ] Verified that the appropriate tests have been written or valid explanation given.
- [ ] If applicable, tested by installing this plugin on the Jenkins instance.

## Why I haven't added unit tests
- I am fairly new to Java, especially in jenkins prespective
- I've added something that goes beyond the plugin concept and may require jenkins API to be able to determine the completeness of task
- I've manually tested and checked in jenkins console and logs are coming up

### CODE REVIEW

**Comment by codeasashu** [comment]
Please add a label Hacktoberfest if you find this little useful

**Comment by nrayapati** [comment]
Thank you for making this PR @codeasashu 

Not sure if I understand what is this code is for?

**Comment by codeasashu** [comment]
@nrayapati First of all, thank you for this wonderful plugin.

When I was using the plugin to initially add a comment to an issue on JIRA, It didn't work (It didn't tell me what I was missing). When I checked my global configuration for JIRA, it seemed correct (test configuration went okay). Hence I wanted to see what was the request and responses are when I was adding a comment. 

I used this function to log the request and response parameters. This provided a more "verbose" output compared to the only "error response and code" which was being returned by default.

TLDR; My intention here is to add a "debug" switch which will allow logging the request and response params as they are being sent via retrofit library. This is just a start in the direction.

**Comment by oleg-nenashev** [comment]
So far it is not clear why it has to be in this plugin

**Comment by codeasashu** [comment]
@oleg-nenashev By default, there is no way to see the verbose output of API requests and responses. I want to contribute so it can be possible to output the params and raw responses to jenkins console output. 

If you feel there is another easier way, please let me know. The only thing I care about is to know why I am unable to add a comment on JIRA when the only response from plugin is:
```
JIRA: Site - SITE NAME - Search JQL: project='PROJ' startAt: 0 maxResults: 3
Error Code: -1
Error Message: null value
```

The way I am doing the debugging is adding `getLogger` to the `JiraService.java` to know what params are being sent, what response I am getting (raw).

**Comment by codeasashu** [comment]
I want to make this clear as well that this is not a complete solution but only a start. Going further, I will:

- add a configuration in the jenkins settings page to allow debug mode
- debug mode will output "verbose" mode in jenkins console, which will include request (with headers and params), response (headers and body (raw)), status code, as well as any data being sent (in json format)

**Comment by oleg-nenashev** [comment]
Thanks for the clarification!

**Comment by codeasashu** [comment]
Now that the logging interceptor is used, I feel the fix done in 80a26bf wasn't necessary. I will remove this after doing some testing.

I am also quite new to jelly standards and still learning, hence I've just dumped retrofit requests to `BODY` level, however, I will add this as a dropdown in jelly for the user to control the debug level.

**Comment by nrayapati** [comment]
@codeasashu Hope you are planning make more changes to this PR, for now I am closing this one as it is been a while since there is update on this PR. Please reopen or create a new one once we have all the changes are done. thank you.
